pep-0712.rst

```diff
@@ -0,0 +1,267 @@
+  @overload
+  def converter(x: bytes) -> list: ... # <- valid
+  def converter(x): ... # __init__ would use argument type 'str | bytes'.
+
+  # The following are invalid converter types.
+  def converter(): ...
+  def converter(**kwargs): ...
+  def converter(x, y): ...
+  def converter(*, x): ...
+  def converter(*args, x): ...
+
+  @overload
+  def converter(): ...
+  @overload
+  def converter(x: int, y: str): ...
+  def converter(x=..., y = ...): ...
+
+  # The following are invalid converter types for a field annotated as type 'list'.
+  def converter(x) -> tuple: ...
+  def converter(x) -> Sequence: ...
+
+  @overload
+  def converter(x) -> tuple: ...
+  @overload
+  def converter(x: int, y: str) -> list: ...
+  def converter(x=..., y = ...): ...
+
+
+Reference Implementation
+========================
+
+The `attrs <#attrs-converters>`_ library already includes a ``converter``
+parameter containing converter semantics.
+
+CPython support can be seen on a branch: `GitHub <#attrs-converters>`_.
+
+Rejected Ideas
+==============
+
+Just adding "converter" to ``dataclass_transform``'s ``field_specifiers``
+-------------------------------------------------------------------------
+
+The idea of isolating this addition to ``dataclass_transform`` was briefly
+discussed in `Typing-sig <#only-dataclass-transform>`_ where it was suggested
+to open this to ``dataclasses``.
+
+Additionally, adding this to ``dataclasses`` ensures anyone can reap the
+benefits without requiring additional libraries.
+
+Automatic conversion using the field's type
+-------------------------------------------
+
+One idea could be to allow the type of the field specified (e.g. ``str`` or
+``int``) to be used as a converter for each argument provided.
+`Pydantic's data conversion <#pydantic-data-conversion>`_ has semantics which
+appear to be similar to this approach.
+
+This works well for fairly simple types, but leads to ambiguity in expected
+behavior for complex types such as generics. E.g. For ``tuple[int]`` it is
+ambiguous if the converter is supposed to simply convert an iterable to a tuple,
+or if it is additionally supposed to convert each element type to ``int``.
+
+Converting the default values
+-----------------------------
+
+Having the synthesized ``__init__`` also convert the default values (such as
+``default`` or the return type of ``default_factory``) when the would make the
+expected type of these parameters complex for type-checkers, and does not add
+significant value.
+
+References
+==========
+.. _#typeshed: https://github.com/python/typeshed
+.. _#attrs-converters: https://www.attrs.org/en/21.2.0/examples.html#conversion
+.. _#cpython-branch: https://github.com/thejcannon/cpython/tree/converter
+.. _#only-dataclass-transform: https://mail.python.org/archives/list/typing-sig@python.org/thread/NWZQIINJQZDOCZGO6TGCUP2PNW4PEKNY/
+.. _#pydantic-data-conversion: https://docs.pydantic.dev/usage/models/#data-conversion
```

Comment:
```suggestion
References
==========

.. _attrs-converters: https://www.attrs.org/en/21.2.0/examples.html#conversion
.. _cpython-branch: https://github.com/thejcannon/cpython/tree/converter
.. _only-dataclass-transform: https://mail.python.org/archives/list/typing-sig@python.org/thread/NWZQIINJQZDOCZGO6TGCUP2PNW4PEKNY/
.. _pydantic-data-conversion: https://docs.pydantic.dev/usage/models/#data-conversion
```

* Typeshed is not mentioned or referenced anywhere
* Remove redundant and potentially confusing `#` in link target names (which looks like autonumbered named footnotes, as well as a URL fragment id)
* Fix line breaks